src/SphereXEngine.sol

```diff
@@ -183,20 +237,31 @@ contract SphereXEngine is ISphereXEngine, AccessControlDefaultAdminRules {
     function _addCfElementFunctionEntry(int256 num) internal {
         require(num > 0, "SphereX error: expected positive num");
         FlowConfiguration memory flowConfig = _flowConfig;
+        EngineConfig memory engineConfig = _engineConfig;
 
         // Upon entry to a new function we should check if we are at the same transaction
         // or a new one. in case of a new one we need to reinit the currentPattern, and save
         // the new transaction "boundry" (block.number+tx.origin+block.timestamp+block.difficulty)
-        bytes3 currentTxBoundaryHash =
-            bytes3(keccak256(abi.encode(block.number, tx.origin, block.timestamp, block.difficulty)));
-        if (currentTxBoundaryHash != flowConfig.txBoundaryHash) {
+        bytes16 currentTxBoundaryHash =
+            bytes16(keccak256(abi.encode(block.number, tx.origin, block.timestamp, block.difficulty)));
+        if (currentTxBoundaryHash != engineConfig.txBoundaryHash) {
             flowConfig.pattern = PATTERN_START;
-            flowConfig.txBoundaryHash = currentTxBoundaryHash;
+            engineConfig.txBoundaryHash = currentTxBoundaryHash;
+            flowConfig.enforce = false;
             if (flowConfig.depth != DEPTH_START) {
                 // This is an edge case we (and the client) should be able to monitor easily.
                 emit TxStartedAtIrregularDepth();
                 flowConfig.depth = DEPTH_START;
             }
```

Comment:
move the comment a bit:

```suggestion
        // Upon entry to a new function we should check if we are at the same transaction or a new one.
        bytes16 currentTxBoundaryHash =
            bytes16(keccak256(abi.encode(block.number, tx.origin, block.timestamp, block.difficulty)));
        if (currentTxBoundaryHash != engineConfig.txBoundaryHash) {
            
            // in case of a new transaction we need to reinit the currentPattern, and save
            // the new transaction "boundry" (block.number+tx.origin+block.timestamp+block.difficulty)
            flowConfig.pattern = PATTERN_START;
            engineConfig.txBoundaryHash = currentTxBoundaryHash;
            flowConfig.enforce = false;
            if (flowConfig.depth != DEPTH_START) {
                // This is an edge case we (and the client) should be able to monitor easily.
                emit TxStartedAtIrregularDepth();
                flowConfig.depth = DEPTH_START;
            }
```